### PR TITLE
GeneticsRefactoring

### PR DESCRIPTION
**Pull request by MrNihil**
## About The Pull Request
Attempt to clean up genetics code and to add functionality that seems that should have been there:
Fixed some variable names
Genetic scanner now properly shows all mutations mob has
Genetic scanner highlights (bolds) active mutations
Made DNA Console description something less generic
Made Genetic Scanner description more straightforward about syncing database
Added option to recycle unused activators

I tried to clean up code a little more but /dna_injector/activator is both a mutator and activator (distinguishable only by /var/research), showing active mutations is hard as we have two lists for default mutations in /mob/dna and a third one for active mutations, that has different structure then other two. Also there was a bug where we could write to mob mutation list from the genetic sequencer. This is quite messy, as you can see from commits I tried to show mutation alias in DNA Console, but reverted that in another commit since it broke the machine.

## Why It's Good For The Game
Some code readability, better descriptions - self-explainatory
Genetic Scanner showing all mutations and highlighting active ones is a quality improvement, allowing to check for active mutations. Removing unused activators should have existed in the code, this also affects mutators (but not map-spawned injectors).

## Changelog
:cl:
fix: You can now recycle unused mutators and activators in dna console.
fix: Genetic scanner now shows all mutations.
expansion: Genetic scanner will highlight active mutations.
/:cl:

### CODE REVIEW

**Comment by MrNihil** [comment]
@MrMelbert @Time-Green 
New PR since the old one went to shit on rebase.

**Comment by tralezab** [comment]
NIHIL! NIHIL! **NIHIIIIIIIIIIIL!!**
image

File: code/game/machinery/computer/dna_console.dm [hunk 1]
```diff
@@ -262,14 +264,14 @@
...
 
-	is_joker_ready = (jokerready < world.time)
-	time_to_joker = round((jokerready - world.time)/10)
+	is_joker_ready = (joker_ready < world.time)
+	time_to_joker = round((joker_ready - world.time)/10)
 
-	is_injector_ready = (injectorready < world.time)
-	time_to_injector = round((injectorready - world.time)/10)
+	is_injector_ready = (injector_ready < world.time)
+	time_to_injector = round((injector_ready - world.time)/10)
```
Comment (SuperNovaa41) [inline code/game/machinery/computer/dna_console.dm:274]: these could probably be done better with our cooldown macros

Comment (MrNihil) [reply code/game/machinery/computer/dna_console.dm:274]: Where can I find examples of these?

**Comment by LemonInTheDark** [comment]
> Added option to recycle unused activators

Doesn't this make printing an activator less of an opportunity cost?

**Comment by MrNihil** [comment]
@LemonInTheDark I dont think so. You still have a cooldown on printing. I thought about reducing this cooldown if you recycle an activator but it gets complicated when someone will just mass print whatever and then bypass the cooldown completely. In this case if you dont want to use an injector and not hiding it or throwing into disposals you can safely destroy it in the console.

**Pull request merged**